### ISSUE TITLE
Transport error in NET6.0

### ISSUE THREAD

**Issue by mezereon-co** [elastic/elasticsearch-net#6052]
**NEST/Elasticsearch.Net version**: 7.15.2

**Elasticsearch version**: 7.15.0

**.NET runtime version**: NET6.0

**Operating system version**: Debian 10 (buster)

**Description of the problem including expected versus actual behavior**:
I upgraded framework from NET5 to NET6 and I noticed intermittent errors (Elasticsearch.Net.Transport) during load test or high traffic. (the same code works perfectly fine in NET5) Below is the error information from the logs:

UnexpectedElasticsearchClientException: Object reference not set to an instance of an object.
  Module "Elasticsearch.Net.Transport`1+<RequestAsync>d__15`1", line 1213, col 0, in MoveNext
    Void MoveNext()
  Module "System.Runtime.ExceptionServices.ExceptionDispatchInfo", line 17, col 0, in Throw
    Void Throw()
  Module "System.Runtime.CompilerServices.TaskAwaiter", line 55, col 0, in ThrowForNonSuccess
    Void ThrowForNonSuccess(System.Threading.Tasks.Task)
  Module "System.Runtime.CompilerServices.TaskAwaiter", line 40, col 0, in HandleNonSuccessAndDebuggerNotification
    Void HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
  Module "Nest.ElasticClient+<SearchAsync>d__266`1", line 0, col 0, in MoveNext
    Void MoveNext()
  Module "System.Runtime.ExceptionServices.ExceptionDispatchInfo", line 17, col 0, in Throw
    Void Throw()
  Module "System.Runtime.CompilerServices.TaskAwaiter", line 55, col 0, in ThrowForNonSuccess
    Void ThrowForNonSuccess(System.Threading.Tasks.Task)
  Module "System.Runtime.CompilerServices.TaskAwaiter", line 40, col 0, in HandleNonSuccessAndDebuggerNotification

**Comment by stevejgordon** [comment]
Hi @mezereon-co. Are you able to share details of the client method(s) being called when these occur? Also, what kind of load is required to start triggering these?

Are you able to [enable DebugMode](https://www.elastic.co/guide/en/elasticsearch/client/net-api/current/debug-mode.html) in your load test environment and capture the DebugInfomation from a failed request, please? It may yield a little more information that could be useful.

**Comment by mezereon-co** [comment]
Steve - thank you for your prompt response and additional questions. There were between 10-50 errors amongst 10,000 requests that were sent in a minute. We re-used Elasticsearch client instance for multiple calls and the call was:
var result = await client.SearchAsync<dynamic>(searchRequest);
Unfortunately, we needed to rollback to NET5.0 for now - we will try to provide more info in the near future

**Comment by mezereon-co** [comment]
I was able to produce more meaningful error messages:

{
"ClassName": "System.NullReferenceException",
"Message": "Object reference not set to an instance of an object.",
"Source": "System.Net.Http",
"StackTraceString": " at System.Net.Http.Headers.HttpHeaders.ReadStoreValues[T](Span`1 values, Object storeValue, HttpHeaderParser parser, Int32& currentIndex) at System.Net.Http.Headers.HttpHeaders.GetStoreValuesAsStringOrStringArray(HeaderDescriptor descriptor, Object sourceValues, String& singleValue, String[]& multiValue) at System.Net.Http.Headers.HttpHeaders.GetStoreValuesAsStringArray(HeaderDescriptor descriptor, HeaderStoreItemInfo info) at System.Net.Http.Headers.HttpHeaders.TryGetValues(HeaderDescriptor descriptor, IEnumerable`1& values) at Elasticsearch.Net.HttpConnection.RequestAsync[TResponse](RequestData requestData, CancellationToken cancellationToken) at Elasticsearch.Net.RequestPipeline.CallElasticsearchAsync[TResponse](RequestData requestData, CancellationToken cancellationToken) at Elasticsearch.Net.Transport`1.RequestAsync[TResponse](HttpMethod method, String path, CancellationToken cancellationToken, PostData data, IRequestParameters requestParameters)",
}

... and 

{
"ClassName": "System.IndexOutOfRangeException",
"Message": "Index was outside the bounds of the array.",
"Source": "System.Net.Http",
"StackTraceString": " at System.Net.Http.Headers.HttpHeaders.ReadStoreValues[T](Span`1 values, Object storeValue, HttpHeaderParser parser, Int32& currentIndex) at System.Net.Http.Headers.HttpHeaders.GetStoreValuesAsStringOrStringArray(HeaderDescriptor descriptor, Object sourceValues, String& singleValue, String[]& multiValue) at System.Net.Http.Headers.HttpHeaders.GetStoreValuesAsStringArray(HeaderDescriptor descriptor, HeaderStoreItemInfo info) at System.Net.Http.Headers.HttpHeaders.TryGetValues(HeaderDescriptor descriptor, IEnumerable`1& values) at Elasticsearch.Net.HttpConnection.RequestAsync[TResponse](RequestData requestData, CancellationToken cancellationToken) at Elasticsearch.Net.RequestPipeline.CallElasticsearchAsync[TResponse](RequestData requestData, CancellationToken cancellationToken) at Elasticsearch.Net.Transport`1.RequestAsync[TResponse](HttpMethod method, String path, CancellationToken cancellationToken, PostData data, IRequestParameters requestParameters)",
}

A quick search on Google pointed me to an issue on NewRelic Github repository: https://github.com/newrelic/newrelic-dotnet-agent/issues/803.  It turned out that NewRelic caused this error and I was able to confirm by deploying the code without NewRelic agent running. Thank you for your time and apologies for a false alert.
